cmd/soroban-cli/src/config/key.rs

```diff
@@ -0,0 +1,148 @@
+use std::{fmt::Display, str::FromStr};
+
+use serde::{Deserialize, Serialize};
+
+use super::secret::{self, Secret};
+use crate::xdr;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Failed to extract public key from secret")]
+    SecretPublicKey,
+    #[error(transparent)]
+    Secret(#[from] secret::Error),
+    #[error(transparent)]
+    StrKey(#[from] stellar_strkey::DecodeError),
+    #[error("failed to parse key {0}")]
```

Comment:
We shouldn't output a key into an error message, we don't know where that is possibly being leaked. Even if it didn't parse.